CDTDatastore/query/CDTQIndex.h

```diff
@@ -40,12 +43,18 @@ extern NSString *const kCDTQTextType;
 
 + (nullable instancetype)index:(NSString *)indexName
                     withFields:(NSArray<NSString *> *)fieldNames
-                        ofType:(NSString *)indexType;
+                        ofType:(NSString *)indexType __deprecated;
+
++ (instancetype)index:(NSString *)indexName
```

Comment:
Might be a good idea to have some doc comments on this method. To be fair, the deprecated one didn't, but it's inconsistent with the others.